src/briefcase/config.py

```diff
@@ -400,9 +417,12 @@ def __init__(
         if not is_valid_app_name(self.app_name):
             raise BriefcaseConfigError(
                 f"{self.app_name!r} is not a valid app name.\n\n"
-                "App names must not be reserved keywords such as 'and', 'for' and 'while'.\n"
-                "They must also be PEP508 compliant (i.e., they can only include letters,\n"
-                "numbers, '-' and '_'; must start with a letter; and cannot end with '-' or '_')."
+                "App names must:\n"
+                "- Not be reserved keywords (like 'and', 'for', 'while', 'main', 'test', etc.)\n"
+                "- Be PEP508 compliant (letters, numbers, hyphens, and underscores only)\n"
```

Comment:
The "PEP508 compliant" restriction isn't correct any more. `hello.world` is valid PEP 508, but it's not a valid app name.